clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3301,14 +3311,24 @@ void Sema::AddSYCLIntelNumSimdWorkItemsAttr(Decl *D,
         return;
       }
     }
-
-    // If the declaration has an [[intel::reqd_work_group_size]] attribute,
-    // check to see if the first argument can be evenly divided by the
+    // If the declaration has a [[intel::reqd_work_group_size()]] or
+    // [[cl::reqd_work_group_size()]] attribute, check to see if the
+    // last argument can be evenly divided by the
     // num_simd_work_items attribute.
+    // If the declaration has a __attribute__((reqd_work_group_size))
+    // attribute, check to see if the first argument can be evenly
+    // divided by the num_simd_work_items attribute.
+    // First and last argument are only swapped for the Intel attributes
+    // and not the OpenCL ones.
     if (const auto *DeclAttr = D->getAttr<ReqdWorkGroupSizeAttr>()) {
       Optional<llvm::APSInt> XDimVal = DeclAttr->getXDimVal(Context);
+      Optional<llvm::APSInt> ZDimVal = DeclAttr->getZDimVal(Context);
+
+      llvm::APSInt WorkGroupSize =
+          DeclAttr->getSyntax() == AttributeCommonInfo::AS_GNU ? *XDimVal
```

Comment:
This would be better done by adding an `Accessor` to the attribute in Attr.td. e.g.,
```
let Accessors = [Accessor<"usesOpenCLArgOrdering", [GNU<"reqd_work_group_size">, CXX11<"cl","reqd_work_group_size">]>];
```
then you can do `DeclAttr->usesOpenCLArgOrdering()` here. That will also fix the issue with the C++ OpenCL spelling.

Reply:
Done